internal/test/framework/windowsvm.go

```diff
@@ -157,21 +158,20 @@ func (w *windowsVM) CopyFile(filePath, remoteDir string) error {
 	return nil
 }
 
-// RetrieveFiles retrieves list of files from remote directory to the local directory.
-// The implementation can be changed if the use-case arises. As of now, we're doing a best effort
-// to collect every log possible. If a retrieval of file fails, we would proceed with retrieval
-// of other log files.
-func (w *windowsVM) RetrieveFiles(remoteDir, localDir string) error {
+// RetrieveDirectories retrieves a whole directory recursively from the remote windows VM to the local/CI directory.
+// This function recursively calls itself if it finds a directory inside, and it calls the retrieveFiles function
+// if it encounters a file.
+func (w *windowsVM) RetrieveDirectories(remoteDir string, localDir string) error {
+
 	if w.sshClient == nil {
-		return fmt.Errorf("RetrieveFile cannot be called without a ssh client")
+		return fmt.Errorf("cannot retrieve remote directory without a ssh client")
 	}
 
-	// Create local dir
+	// Create a local dir
 	err := os.MkdirAll(localDir, os.ModePerm)
 	if err != nil {
-		log.Printf("could not create %s: %s", localDir, err)
+		return fmt.Errorf("could not create %s: %v", localDir, err)
```

Comment:
In case the `os.MkdirAll` fails because directory already exists, can we just append the contents (or replace the existing logs)?

Reply:
I can log the error instead, so it passes to the next step. Even if the directory exists, retrieveFiles can copy the file to already existing local directory.

Reply:
This change is not required as os.MkdirAll returns nil(no err) if the directory already exists.